docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+            raise e
+        logger.info("Vectorization Layer complete...")
+        try:
+            def vectorize_text(text, label):
+                text = tf.expand_dims(text, -1)
+                return vectorize_layer(text), label
+
+            train_ds = raw_train_ds.map(vectorize_text)
+            val_ds = raw_val_ds.map(vectorize_text)
+            test_ds = raw_test_ds.map(vectorize_text)
+            logger.info("Dataset mapped to text")
+            AUTOTUNE = tf.data.AUTOTUNE
+            train_ds = train_ds.cache().prefetch(buffer_size=AUTOTUNE)
+            test_ds = test_ds.cache().prefetch(buffer_size=AUTOTUNE)           
+            val_ds = val_ds.cache().prefetch(buffer_size=AUTOTUNE)
+            logger.info("Dataset cached")
+            model = tf.keras.Sequential([
+                layers.Embedding(max_features + 1, embedding_dim),
+                layers.Dropout(0.2),
+                layers.GlobalAveragePooling1D(),
+                layers.Dropout(0.2),
+                layers.Dense(1)])
+            model.compile(loss=losses.BinaryCrossentropy(from_logits=True),
+                optimizer='adam',
+                metrics=tf.metrics.BinaryAccuracy(threshold=0.0))
+            logger.info("beginning training...")
+            history = model.fit(train_ds, validation_data=val_ds,
+                    epochs=epochs)
+            mp_queue.put(history)
+            logger.info("Saving...")
+            model.save(model_save_loc, save_format='tf', overwrite=True)
+        except Exception as e:
+            logger.error(f"In precompile {e}")
+
+        try:
+            logger.info(f"Training complete!")
+            export_model = tf.keras.Sequential([
+                vectorize_layer,
+                model,
+                layers.Activation('sigmoid')
+                ])
+            logger.info(f"Compiling new model...")
+            export_model.compile(
+                loss=losses.BinaryCrossentropy(from_logits=False), optimizer="adam", metrics=['accuracy']
+            )
+            loss, accuracy = model.evaluate(test_ds)
+            logger.info(f"Evaluation Results: \n {str(loss)} \n {str(accuracy)}")
+            export_model.save(f"{model_save_loc}-exported", save_format="tf", overwrite=True)
+        except Exception as e:
+            logger.error(f"ERR in TRAIING \n {e}")
+            raise e
+```
+
+While this method may be intimidating, don't let it scare you! It is simply the core code of our companion piece, gently refactored to fit into our tool's (and the PythonSdk's) runtime and lifecycle.
+
+In this method we:
+  * load our dataset `tf.data.Dataset.save's sibling
+  * create and adapt our `vectorize_layer: tensorflow.keras.layers.TextVectorization` as recommended by TensorFlow
+  * cache our datasets (now loading data as `Datasets` begins to pay off!)
+  * init, compile, evaluate, and save our base model.]
+  * generate  data for our previews
```

Comment:
TW review:

In this method we...
  * Load the dataset `tf.data.Dataset.save's sibling.
  * Create and adapt the `vectorize_layer: tensorflow.keras.layers.TextVectorization` as recommended by TensorFlow.
  * Cache the datasets (now loading data as `Datasets` begins to pay off!)
  * Init, compile, evaluate, and save the base model.
  * Generate data for the previews.